src/utils/batch_group.hpp

```diff
@@ -0,0 +1,23 @@
+// Copyright 2025 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef SRC_UTILS_BATCH_GROUP_HPP
+#define SRC_UTILS_BATCH_GROUP_HPP
+
+#include "skity/geometry/matrix.hpp"
+#include "skity/graphic/paint.hpp"
+#include "src/utils/array_list.hpp"
+
+namespace skity {
+
+template <typename T>
+struct BatchGroup {
+  std::vector<T> items;
+  std::vector<Paint> paints;
+  std::vector<Matrix> transforms;
```

Comment:
Would it be better to manage these objects with a vector? For example, when a vector grows, it only needs to allocate memory once instead of three times, and it would also be easier to modify the code in the future (for example, we may change Matrix to a matrix ID later and put the matrices into an SSBO).